splinterd/src/config/partial.rs

```diff
@@ -14,11 +14,22 @@
 
 use std::time::Duration;
 
+/// ConfigSource displays the source of configuration values, used to identify which of the various
+/// config modules were used to create a particular PartialConfig object.
+#[derive(Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub enum ConfigSource {
+    Toml { file: String },
+    Default,
```

Comment:
Is `Some(ConfigSource::Default)` different from `None`? Maybe we could scrap the `Option` if `PartialConfig` is constructed with a `new(source: ConfigSource) -> Self` method rather than `default()`, which would simplify some things.

Reply:
Yes, the Default is specifically from the DefaultConfig module.

Reply:
I addressed this by changing the PartialConfig source field to `ConfigSource` from `Option<ConfigSource>`